src/RankMovie.jsx

```diff
@@ -38,11 +53,46 @@ function RankMovie() {
     }
   }, [low, high, newMovie]);
 
-  const handleComparison = (prefersNewMovie) => {
-    if (prefersNewMovie) {
-      setLow(mid + 1);
-    } else {
-      setHigh(mid - 1);
+  function updateMovieRatings(movieId, rating) {
+    const userId = currentUser.uid;
+    fetch("http://localhost:3000/users/logMovie", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ userId, movieId, rating }),
+    })
+      .then((response) => response.json())
+      .catch((error) => {
+        console.error("Error logging movie:", error);
+      });
+  }
+
+  const handleComparison = async (prefersNewMovie) => {
```

Comment:
I think this calculations should be done in the backend. All you need is an API endpoint with the params for the two movies and the prefers flag.

Reply:
Now that I changed the algorithm I'm not sure that I can, or it might be more complicated. The new rating is based on multiple other movies ratings so those would have to be inputs as well. What do you think?

Reply:
It's fine to take more input, it can be an array of IDs for instance. Or an array of objects with details as long as the usual payload is not too big.

It's always recommended to keep heavy calculations in the backend. 
It seems to me that this algorithm is one of the main calculations your app will do.

Reply:
@kevinzhu00 @samyalihamad Do you have thoughts on this matter?

Reply:
I initially would have said it was ok to keep here if it was a small calculation because the backend call has its overhead as well. If the rating is taking multiple other movies than that isn't really scalable and the backend api makes more sense. You can add it the backend as a stretch goal after completing TC1 and TC2.